### ISSUE TITLE
Smartlinking of discussion participants doesn't work

### ISSUE THREAD

**Issue by stadtwald** [loomio/loomio#1767]
When viewing a discussion, mentioning a discussion participant results in a link being displayed.  This link simply references a location on the same page via a hash (#<participantname>).  When clicked, a Javascript event is fired, but that seems to fail with "this.query is null" on my browser (Iceweasel 31 on Debian Wheezy).  The result is that nothing visible happens.

**Comment by denjello** [comment]
Crazy as in WTF?

Here is the valid mention_user.feature

```
Scenario: User views a comment with a mention
  Then we should make this test work
  Given "Harry" is a member of the group
  And a comment exists mentioning "@harry"
  When I visit the discussion page
  Then I should see a link to "harry"s user
```

`discussions_helper.rb`

```
  def add_mention_links(comment)
    auto_link_usernames_or_lists(comment, :username_url_base => "#", :username_include_symbol => true)
  end
```

`#harry` is not a link to "harry"s user.

**Comment by denjello** [comment]
I found a solution to this, which is pretty hacky because I am recycling an existing route. Here is how it works: When an `@denjello` mention is found in the comment it creates a link to `/u/denjello` which, when routed, is still parsed by users_controller.rb because it naturally couldn't find the ID and falls back to the other option. Thankfully the system prevents the occurance of the same username more than once, so this approach could work in production.

`app/helpers/discussions_helper.rb`
```
    auto_link_usernames_or_lists(comment, :username_url_base => "/u/", :username_include_symbol => true)
```

`app/serializers/comment_serializer.rb`
```
    auto_link_usernames_or_lists object.body, username_url_base: '/u/', username_include_symbol: true

```

`app/controllers/users_controller.rb`
```
    @user = User.find_by_key(params[:id]) || User.find_by_username!(params[:id])
```

**Comment by denjello** [comment]
And from a useability perspective, I find the link `/u/denjello` quite a bit nicer than `/u/TDi0YBx5/denjello` because I can at least remember the first one and even use that structure to find out more about someone else (who is in a group I am also in).

**Comment by robguthrie** [comment]
resolved in 1.0
